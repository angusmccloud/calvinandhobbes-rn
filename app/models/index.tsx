import {eIcons} from 'models/icons/icons';

// Export eNums
export {eIcons};

// Export Interfaces
