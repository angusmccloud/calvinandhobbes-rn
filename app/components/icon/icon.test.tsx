import React from 'react';
import renderer from 'react-test-renderer';
import {eIcons} from 'models';
import Icon from './icon';

describe('Icon Component rendered correctly', () => {
  it('tab One Icon', () => {
    const icon = renderer.create(<Icon icon={eIcons.tabOne} />);
    expect(icon).toMatchSnapshot();
  });
  it('tab One Focused Icon', () => {
    const icon = renderer.create(<Icon icon={eIcons.tabOneFocused} />);
    expect(icon).toMatchSnapshot();
  });
  it('tab Two Icon', () => {
    const icon = renderer.create(<Icon icon={eIcons.tabTwo} />);
    expect(icon).toMatchSnapshot();
  });
  it('tab Two Focused Icon', () => {
    const icon = renderer.create(<Icon icon={eIcons.tabTwoFocused} />);
    expect(icon).toMatchSnapshot();
  });
  it('tab Three Icon', () => {
    const icon = renderer.create(<Icon icon={eIcons.tabThree} />);
    expect(icon).toMatchSnapshot();
  });
  it('tab Three Focused Icon', () => {
    const icon = renderer.create(<Icon icon={eIcons.tabThreeFocused} />);
    expect(icon).toMatchSnapshot();
  });
  it('tab Four Icon', () => {
    const icon = renderer.create(<Icon icon={eIcons.tabFour} />);
    expect(icon).toMatchSnapshot();
  });
  it('tab Four Focused Icon', () => {
    const icon = renderer.create(<Icon icon={eIcons.tabFourFocused} />);
    expect(icon).toMatchSnapshot();
  });
  it('About Icon', () => {
    const icon = renderer.create(<Icon icon={eIcons.about} />);
    expect(icon).toMatchSnapshot();
  });
  it('About Focused Icon', () => {
    const icon = renderer.create(<Icon icon={eIcons.aboutFocused} />);
    expect(icon).toMatchSnapshot();
  });
});
