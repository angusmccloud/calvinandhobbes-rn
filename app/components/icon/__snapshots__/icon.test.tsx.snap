// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component rendered correctly About Focused Icon 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#0f348e",
          "fontSize": 50,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Icon Component rendered correctly About Icon 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#0f348e",
          "fontSize": 50,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Icon Component rendered correctly Backspace Icon 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#0f348e",
          "fontSize": 50,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Icon Component rendered correctly Minus Icon 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#0f348e",
          "fontSize": 50,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Icon Component rendered correctly Plus Icon 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#0f348e",
          "fontSize": 50,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Icon Component rendered correctly Task Four Focused Icon 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#0f348e",
          "fontSize": 50,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Icon Component rendered correctly Task Four Icon 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#0f348e",
          "fontSize": 50,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Icon Component rendered correctly Task One Focused Icon 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#0f348e",
          "fontSize": 50,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Icon Component rendered correctly Task One Icon 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#0f348e",
          "fontSize": 50,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Icon Component rendered correctly Task Three Focused Icon 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#0f348e",
          "fontSize": 50,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Icon Component rendered correctly Task Three Icon 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#0f348e",
          "fontSize": 50,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Icon Component rendered correctly Task Two Focused Icon 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#0f348e",
          "fontSize": 50,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Icon Component rendered correctly Task Two Icon 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#0f348e",
          "fontSize": 50,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;
